universal-application-tool-0.0.1/app/models/Program.java

```diff
@@ -39,17 +42,20 @@ public Program(ProgramDefinition definition) {
     this.programDefinition = definition;
   }
 
-  /** Populates column values from {@link programDefinition} */
+  /** Populates column values from {@link ProgramDefinition} */
   @PrePersist
+  @PreUpdate
   public void serializeBlockDefinitions() throws JsonProcessingException {
     this.id = this.programDefinition.id();
     this.name = this.programDefinition.name();
     this.description = this.programDefinition.description();
     this.blockDefinitions = mapper.writeValueAsString(this.programDefinition.blockDefinitions());
   }
 
-  /** Populates {@link programDefinition} from column values. */
+  /** Populates {@link ProgramDefinition} from column values. */
   @PostLoad
+  @PostPersist
```

Comment:
question: What is the difference between @PostLoad and @PostPersist and why do we need both? Could we document that decision?

Reply:
Load - read from DB
Persist - save new row to DB
Update - update existing row to DB

We _could_ have a separate one for `@PostPersist` since that one won't need to `setBlockDefinitions` but I think it's cleaner if we just have one for all `@PostX`.